### PR TITLE
Add recipe for building Triangle mesh generator

### PR DESCRIPTION
**Pull request by j-fu**
Hi, this should replace JuliaGeometry/TriangleBuilder.

### CODE REVIEW

File: T/Triangle/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# https://github.com/JuliaGeometry/Triangulate.jl
+#
+# Please be aware that triunsuitable.c is not part of the original Triangle distribution.
+# It provides the possibility to pass a cfunction created in Julia as  user refinement callback.
+# For this reason at least triunsiutable.c must be downloaded from Triangulate.jl repo, and
+# for simplicity, we do this for the whole code.
+# We also patch the code in order to replace exit() calls in the C code  so that we can
+# throw an error instead.
+sources = [
+    GitSource("https://github.com/JuliaGeometry/Triangulate.jl.git","b2ffb23ca7d89c567fd31367882bd216757cdb9c")
```
Comment (giordano) [inline T/Triangle/build_tarballs.jl]: Isn't there an upstream source that we can use here?

Comment (j-fu) [reply T/Triangle/build_tarballs.jl]: There is:
http://www.netlib.org/voronoi/triangle.zip
In fact I think this is quite stable. If you agree I'll try to update this.

Comment (j-fu) [reply T/Triangle/build_tarballs.jl]: However, I need to mix this with another file which I would like to keep taking from Triangulate.jl

Comment (giordano) [reply T/Triangle/build_tarballs.jl]: Uhm, the only problem of that link is that it isn't versioned -- i.e., it isn't like `triangle-1.6.zip` -- so they may overwrite it in the future, as unlikely as it is :persevere:  Any other option?

Comment (j-fu) [reply T/Triangle/build_tarballs.jl]: AFAIK no. I just double checked, but it's not there.
Netlib itself is appears to be quite stable, and  as far as I know the author (personally) I think indeed that he won't develop it any further, besides of bug fixes. 

... I see the merge ;) , so keeping it in Triangulate appears to be the best...

Comment (giordano) [reply T/Triangle/build_tarballs.jl]: Yeah, I don't think there is an ideal solution here, so this was good enough :slightly_smiling_face:

**Pull request merged**